sendmail/datadog_checks/sendmail/sendmail.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+from datadog_checks.base import AgentCheck, ConfigurationError, is_affirmative
+from datadog_checks.utils.subprocess_output import get_subprocess_output
+
+
+class SendmailCheck(AgentCheck):
+
+    SERVICE_CHECK_NAME = 'sendmail.returns.output'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        if instances is not None and len(instances) > 1:
+            raise ConfigurationError('Sendmail check only supports one configured instance.')
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances=instances)
+
+    def check(self, instance):
+        (sendmail_command, use_sudo, tags) = self._get_config(instance)
+
+        if not sendmail_command:
+            raise ConfigurationError('Please provide the sendmail command in the configuration')
+
+        valid_commands = ["mailq", "sendmail"]
+
+        if not any(cmd in sendmail_command for cmd in valid_commands):
+            raise ConfigurationError("{} does not seem to be a valid command".format(sendmail_command))
+
+        try:
+            queue_size = self._get_sendmail_stats(sendmail_command, use_sudo)
+            self.gauge('sendmail.queue.size', queue_size, tags=tags + ['queue:total'])
+            self.log.debug("Sendmail queue size: {} mails".format(queue_size))
+            # Send an OK service check as well
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags, message='Sendmail OK')
+        except OSError as e:
+            self.log.info("Cannot get sendmail queue info".format(str(e)))
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags, message=str(e))
+
+    def _get_config(self, instance):
+        sendmail_command = instance.get('sendmail_command')
+        use_sudo = is_affirmative(instance.get('use_sudo', False))
+        tags = instance.get('tags', [])
+
+        return sendmail_command, use_sudo, tags
+
+    def _get_sendmail_stats(self, sendmail_command, use_sudo):
+
+        if not os.path.exists(sendmail_command):
+            raise Exception('{} does not exist'.format(sendmail_command))
+
+        self.log.debug(sendmail_command)
+
+        # mailq sample output
+        """
+        MSP Queue status...
+        /var/spool/mqueue-client is empty
+            Total requests: 0
+        MTA Queue status...
+        /var/spool/mqueue is empty
+            Total requests: 0
+        """
+
+        # if we want to use sendmail, we need to append -bp to it
+        # https://www.electrictoolbox.com/show-sendmail-mail-queue/
+        if "sendmail" in sendmail_command:
+            command = [sendmail_command, '-bp']
+        else:
+            command = [sendmail_command]
+
+        # Listing the directory might require sudo privileges
+        if use_sudo:
+            test_sudo = os.system('setsid sudo -l < /dev/null')
```

Comment:
Let's just try it instead like https://github.com/DataDog/integrations-core/blob/712bbee9135f5ac0a05f830d86f188cadb287ee1/process/datadog_checks/process/process.py#L201